iceoryx2/src/service/attribute.rs

```diff
@@ -207,7 +207,12 @@ impl AttributeVerifier {
 
     /// Requires that a specific key is defined.
     pub fn require_key(mut self, key: &str) -> Self {
-        self.required_keys.push(key.into());
+        self.required_keys.push(
+            fatal_panic!(
```

Comment:
This is bad. In theory, it could come from some kind of user input and would lead to a fatal crash.

The option is to not use `&str` for the key argument but the `AttributeKeyString` type. The user can then use the function like:
```rust
Attribute.require_key("my key value".try_from()?);
```